src/models/StyledComponent.js

```diff
@@ -43,162 +43,174 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
       : componentId
   }
 
-  class BaseStyledComponent extends Component {
-    static target: Target
-    static styledComponentId: string
-    static attrs: Object
-    static componentStyle: Object
-    static warnTooManyClasses: Function
-
-    attrs = {}
-    state = {
-      theme: null,
-      generatedClassName: '',
-    }
-    unsubscribeId: number = -1
-
-    unsubscribeFromContext() {
-      if (this.unsubscribeId !== -1) {
-        this.context[CHANNEL_NEXT].unsubscribe(this.unsubscribeId)
+  const createBaseStyledComponent = (options: Object) => {
```

Comment:
The primary change here is wrapping the BaseStyledComponent in a function so that the options can be passed through to `generateAndInjectStyles`. For a clearer diff, I suggest looking at this one with `?w=1`.